src/main/java/org/breedinginsight/brapps/importer/services/FileImportService.java

```diff
@@ -586,8 +571,12 @@ public List<ImportMapping> getSystemMappingByName(String name) {
         return importMappings;
     }
 
-    public List<ImportMappingWorkflow> getWorkflowsForSystemMapping(UUID mappingId) {
-        return importMappingWorkflowDAO.getWorkflowsByImportMappingId(mappingId);
+    public List<ImportWorkflow> getWorkflowsForSystemMapping(UUID mappingId) throws DoesNotExistException {
+        ImportMapping mappingConfig = importMappingDAO.getMapping(mappingId)
+                .orElseThrow(() -> new DoesNotExistException("Cannot find mapping config associated with upload."));
+        BrAPIImportService importService = configManager.getImportServiceById(mappingConfig.getImportTypeId())
+                .orElseThrow(() -> new DoesNotExistException("Config with that id does not exist"));
+        return importService.getWorkflows();
     }
```

Comment:
I agree that there probably won't be a use case for getting all workflows for all services, but it could be implemented readily by reading all mappingIds from the database and calling this method on each one in a loop.